ctapipe/tools/tests/test_tools.py

```diff
@@ -252,6 +277,75 @@ def test_stage_1(tmpdir):
             assert "peak_time" in dl1_image.dtype.names
 
 
+def test_stage_1_dl1(tmpdir, dl1_image_file, dl1_parameters_file):
+    from ctapipe.tools.stage1 import Stage1Tool
+
+    config = Path("./examples/stage1_config.json").absolute()
+    # DL1A file as input
+    with tempfile.NamedTemporaryFile(suffix=".hdf5") as f:
```

Comment:
you should not use tempfile.NamedTemporaryFile here, it's better to just use the tmp_dir fixture you are using (the other tests also misuse this a lot as well I notice).  
That ensures the files are written to the temporary directory defined by the *test system*, not some other place (which is confusing when debugging).   Also you don't need to open the files as a filehandle, as NamedTemporaryFile does, only generate a name.  Note that tmpdir is unique for all test runs, so the actual filename does not need to be random at all, you can pick something easy to read like "test_dl1a_file.hdf5":

For example, just do:
```py3
filename = tmp_dir /  "test_dl1a_file.hdf5"
```
Then later just use `f"some string {filename}"` (not `f.name`)

No "with" or anything else is neeeded.  Then

Reply:
That is also much shorter
I copied that from other tests. so its probably the best if I fix them as well.